types/vee-validate.d.ts

```diff
@@ -235,18 +520,30 @@ export class ValidationProvider extends Vue {
     setFlags(value: { [x: string]: boolean }): void;
 }
 
-export const withValidation: (component: Vue, mapFn: (ctx: ValidationSlotScopeData) => Object) => Vue;
-
-export const version: string;
-
-export const install: Vue.PluginFunction<Configuration>
-
-export const directive: Vue.DirectiveOptions;
+export default class VeeValidate {
+    readonly config: Configuration;
+    readonly i18nDriver: IDictionary;
+    static readonly i18nDriver: IDictionary;
+    static readonly config: Configuration;
+    static version: string;
+    static install: PluginFunction<Configuration>;
+    static Validator: Validator;
+    static ErrorBag: ErrorBag;
+    static Rules: Rules;
+    static ValidationProvider: ValidationProvider;
+    static ValidationObserver: ValidationObserver;
+    static mixin: VueConstructor | ComponentOptions<Vue>;
+    static directive: DirectiveOptions;
 
-export const Rules: {
-    [key: string]: {
-        validate(value: any, args: Object | any[], data?: any): boolean | Promise<boolean>;
-        options?: object;
-        paramNames?: string[];
-    }
+    configure(cfg: Configuration): void;
+    static setI18nDriver(driver: string, instance: IDictionary): void;
+    static configure(cfg: Configuration): void;
+    static setMode(mode: Modes, implementation: Function): void;

```

Comment:
Doesn't this disallow adding new modes? for example:

```js
VeeValidate.setMode('myMode', () => {});
```

Reply:
Yes, i have fix that.